### PR TITLE
PP-225: Hapi v17, Node.js v14, update deps, nyc coverage, modernize

### PR DESCRIPTION
**Pull request by tlhunter**
- update packages
- istanbul to nyc
- test on Node.js v14
- Hapi v17 compatibility
- updated linter rules
- SemVer MAJOR change

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,7 +11,7 @@ pids
 lib-cov
 
 # Coverage directory used by tools like istanbul
```
Comment (Lianathanoj) [inline .gitignore]: super nit: if we're deprecating istanbul, we can say `nyc` instead

File: .node-version [hunk 2]
```diff
@@ -1 +1 @@
-8.11.3
\ No newline at end of file
+14.15.5
```
Comment (Lianathanoj) [inline .node-version]: i believe some of our main repos are in `14.17.0` now, so we can bump this up a bit more

File: lib/index.js [hunk 3]
```diff
@@ -124,13 +115,8 @@ exports.register = (server, options, next) => {
...
   });
-
-  next();
-
 };
 
-exports.register.attributes = {
-  name: 'hapi-rate-limiter'
-};
+exports.pkg = require('../package.json');
```
Comment (Lianathanoj) [inline lib/index.js:122]: you can also specify `exports.name = 'hapi-rate-limiter'` if you want parity

**Review by Lianathanoj — APPROVED** [review]
few nits but LGTM!

File: lib/index.js (hunk 3 above)
Comment (tlhunter) [reply lib/index.js:122]: the `.pkg.name` and `.pkg.version` get hoisted up to `.name` and `.version`

Comment (Lianathanoj) [reply lib/index.js:122]: oh that's awesome!

**Pull request merged**